mcp/setup.sh

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env bash
+set -o pipefail
+
+entry(){
+  echo ""
+  echo "Multi-Cloud-Platform Setup"
+  echo ""
+  echo "Select Option: "
+
+  select choice in "setup" "get" "destroy" "help" "exit"; do
+    case $choice in 
+      setup )
+        installDependencies
+        runSetup
+        ;;
+      get )
+        installDependencies
+        getState
+        ;;
+      destroy )
+        installDependencies
+        runDestroy
+        ;;
+      help) help;;
+      exit) exit;;
+      *) exit;;
+    esac
+  done
+  exit
+}
+
+help() {
+  echo "Usage: ${SCRIPT_NAME} <command> [options]"
+  echo ""
+  echo "Commands:"
+  echo "  setup      Setup Terraform service deployment"
+  echo "  get        Get Terraform state for current direectory"
+  echo "  destroy    Destroy Terraform infrastructure"
+  echo ""
+  entry
+}
+
+
+installDependencies() {
+  echo ""
+  echo "Installing dependencies"
+
+    OS="$(uname)"
+    case ${OS} in
+        'Linux')
+            installLinuxDependencies
+            ;;
+        'Darwin')
+            installDarwinDependencies
+            ;;
+        *)
+            echo "Couldn't determine OS type."
+            return
+            ;;
+    esac
+}
+
+installLinuxDependencies() {
+  if  ! [ -x "$(command -v terraform)" ]
+  then
+    echo ""
+    echo "Getting terraform ..."
+    echo ""
+
+    cd "${BIN}"
+    TERRAFORM_URL_LIN=https://releases.hashicorp.com/terraform/0.12.0/terraform_0.12.0_linux_amd64.zip
+    TERRAFORM_FILE_LIN="${TERRAFORM_URL_LIN##*/}"
+    wget "${TERRAFORM_URL_LIN}"
+    unzip -o "${TERRAFORM_FILE_LIN}"
+    rm "${TERRAFORM_FILE_LIN}"
+    cd "${SCRIPT_DIR}"
+
+    echo ""
+    echo "Terraform for $OS installed."
+    echo ""
+  else
+    echo ""
+    echo "All dependencies installed"
+    echo ""
+  fi
+}
+
+installDarwinDependencies(){
+   # Install Terraform
+    if  ! [ -x "$(command -v terraform)" ]
+    then
+        echo ""
+        echo "Getting terraform ..."
+        echo ""
+
+        cd "${BIN}"
+        TERRAFORM_URL_DAR=https://releases.hashicorp.com/terraform/0.12.0/terraform_0.12.0_linux_amd64.zip
+        TERRAFORM_FILE_DAR="${TERRAFORM_URL_DAR##*/}"
+
+        echo "URL: ${TERRAFORM_URL_DAR}"
+        echo "File: ${TERRAFORM_FILE_DAR}"
+
+        curl "${TERRAFORM_URL_DAR}" --output "${TERRAFORM_FILE_DAR}"
+        unzip "${TERRAFORM_FILE_DAR}"
+        rm "${TERRAFORM_FILE_DAR}"
+        cd "${SCRIPT_DIR}"
+
+        echo ""
+        echo "Terraform for $OS installed."
+        echo ""
+    fi
+}
+
+runSetup() {
```

Comment:
don't put the deployment into the setup function. This should be just creating the local module file and initialising. We have a deployment tool already but I still think it is useful to have a basic version of it for this purpose, so put a `describe` argument for the plan output and a `deploy` argument for the apply.

```
terraform {
  backend "gcs" {
    bucket  = "BUCKET_NAME"
    prefix  = "FOLDER_NAME"
  }
}

module "deployment" {
  source = "github.com/mesoform/terraform-infrastructure-modules/mcp"
}
```